addons/mail/static/src/js/owl/component/chat_window.js

```diff
@@ -80,6 +80,19 @@ class ChatWindow extends owl.store.ConnectedComponent {
         return this.state.folded;
     }
 
+    getState(){
+        const {
```

Comment:
shorter:
```js
/**
 * @return {Object}
 */
getState() {
   return {
      ...this.refs.thread.getComposerState();
      scrollTop: this.refs.thread.getScrollTop();
   };
}
```

Reply:
Also, this method makes no sense for chat windows without any thread. Should put a guard and return undefined

Reply:
I need to add "composer" before the keys of getComposerState so I think the shorter version won't work or am I missing something ?

Reply:
```js
const {
   attachmentLocalIds: composerAttachmentLocalIds,
   textInputHtmlContent: composerTextInputHtmlContent,
} = this.refs.thread.getComposerState();
return {
   composerAttachmentLocalIds,
   composerTextInputHtmlContent,
   scrollTop: this.refs.thread.getScrollTop();
};
```

Reply:
Didn't about that way of changing key when decomposing an object. I find this a little bit confusing as it is the exact opposite of assignation in an object but I'll get used to it :p !